### PR TITLE
Add repo files

### LINKED ISSUE #455: Update repo with latest dev guide recommendations
- [ ] Add missing required files
- [ ] Use pre-commit

### PR DESCRIPTION
**Pull request by Andy-Grigg**
Closes #455 

Adds additional required files to the repository root. Also back-populates the changelog.

### CODE REVIEW

**Comment by Andy-Grigg** [comment]
Not sure about all the warnings from codacy, but these are standard files across the org, so there's not much we can do here.

File: AUTHORS.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# This is the list of GRANTA MI RecordLists API significant contributors.
```
Comment (ludovicsteinbach) [inline AUTHORS.md:1]: Needs updating

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Contributor Covenant Code of Conduct
```
Comment (ludovicsteinbach) [inline CODE_OF_CONDUCT.md:1]: It's weird that it's recommended to add this file when the code of conduct seems to be "inherited" from the org https://github.com/ansys/openapi-common?tab=coc-ov-file#contributor-covenant-code-of-conduct.

**Review by ludovicsteinbach — APPROVED** [review]
All good.

**Pull request merged**